bliss/models/galsim_encoder.py

```diff
@@ -0,0 +1,347 @@
+from pathlib import Path
+from typing import Optional, Tuple, Union
+
+import numpy as np
+import pytorch_lightning as pl
+import torch
+from einops import rearrange
+from matplotlib import pyplot as plt
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+from torch import Tensor, nn
+from torch.distributions import LogNormal, LogisticNormal, Normal
+from torch.nn import functional as F
+from torch.nn.utils import weight_norm
+from torch.optim import Adam
+
+from bliss.catalog import TileCatalog, get_images_in_tiles, get_is_on_from_n_sources
+from bliss.models.decoder import ImageDecoder, get_mgrid
+from bliss.models.galaxy_encoder import CenterPaddedTilesTransform
+from bliss.models.detection_encoder import EncoderCNN, make_enc_final
+
+def get_galsim_params_nll(galaxy_bools, true_params, params):
+    # return shape (n_ptiles x max_detections x max_detections)
+    assert true_params.shape[:-1] == params.shape[:-1]
+
+    galaxy_bools = galaxy_bools.view(-1)
+    true_params = true_params.view(-1, true_params.shape[-1]).transpose(0, 1)
+    params = params.view(-1, params.shape[-1]).transpose(0, 1)
+
+    # only compute loss where lensing is present
+    true_params = true_params[:, galaxy_bools > 0]
+    params = params[:, galaxy_bools > 0]
+
+    # if no lenses present, skip loss calculation
+    log_prob = torch.zeros(1, requires_grad=True).to(device=true_params.device)
+    if true_params.shape[-1] == 0:
+        return -log_prob
```

Comment:
Not sure I understand this, don't you want the galaxy encoder to work even if no lenses are present?